src/directory.c

```diff
@@ -0,0 +1,8 @@
+#include "directory.h"
+
+const char	*directory_get_current_dir(void)
+{
+	char	buffer[BUFFER_SIZE];
+
+	return (getcwd(buffer, sizeof(buffer)));
```

Comment:
compile this .c file and you'll see why that code will not work. you're writing to stackmemory which is popped when the function returns. if you want to return a memory ptr, malloc it.

#include "string.h"

char *f()
{
	char buf[128];
	memset(buf, 'A', 128);
	return (buf);
}

int main()
{
	char *buf = f();
	return (buf != NULL);
}